docs/development/remote-debugging-golang-on-kubernetes.md

```diff
@@ -0,0 +1,123 @@
+# Remote Debugging Golang-Based Services on Kubernetes with Delve and VS Code
+
+This guide walks you through setting up a remote debugging environment for Go applications deployed on a Kubernetes cluster using Delve and VS Code. The steps apply to any Go service on Kubernetes, using the Scality COSI driver as an example.
+
+---
+
+## Prerequisites
+
+Ensure the following are installed and configured before starting:
+
+1. **Kubernetes Cluster**: A running Kubernetes cluster, such as [Minikube](https://minikube.sigs.k8s.io/docs/start/), [Docker Desktop Kubernetes](https://docs.docker.com/desktop/features/kubernetes/), or a remote cluster.
+2. **kubectl**: [Installed](https://kubernetes.io/docs/tasks/tools/) and configured for your cluster.
+3. **VS Code**: [Installed](https://code.visualstudio.com/), with the [Go extension](https://marketplace.visualstudio.com/items?itemName=golang.Go).
+4. **COSI Driver**: Clone the repository and navigate to the directory:
+
+   ```bash
+   git clone git@github.com:scality/cosi-driver.git && cd cosi-driver
+   ```
+
+5. **Delve**: [Installed locally](https://github.com/go-delve/delve/tree/master/Documentation/installation).
+
+---
+
+## Step 1: Prepare the Dockerfile for Delve Debugging
+
+To enable remote debugging with Delve, modify the Dockerfile to include Delve in a dedicated development stage. This configuration ensures Delve runs in headless mode, ready for remote debugging.
+
+Replace the default Dockerfile with one that supports Delve:
+
+```bash
+mv ./Dockerfile ./Dockerfile.bkp && cp docker/delve/Dockerfile ./
+```
+
+---
+
+## Step 2: Build the Container Image
+
+Build the Docker image with Delve by running:
+
+```bash
+make container
```

Comment:
Alternatively, just make a `delve-container` make command? Since there are two Dockerfiles, I think it makes sense to have two build options. Your call.

Reply:
I took your ssuggestion, thanks
Keeps debug and prod env seperate